packages/hooks/src/useDocumentListener.ts

```diff
@@ -0,0 +1,29 @@
+import { useEffect } from 'react';
+
+/**
+ * Hook to subcribe to an event listener.
+ * @param type Represents the event type to listen for.
+ * @param eventCallback Event listener callback function.
+ * @param enabled Determines whether or not the hook should run, defaults to true.
+ * @param options Optional parameter to specify options passed to the eventListener.
+ * @param dependencies Optional array to be passed to useEffect hook, such that the hook will only run if the array's values have changed.
+ */
+export default function useDocumentEventListener(
+  type: string,
+  eventCallback: EventListener,
+  enabled: boolean = true,
```

Comment:
I can't decide if this should be the last argument, or second to last. I feel like last as it's most likely that you won't need to set this to false.

Reply:
Moved to last arg!